main.tf

```diff
@@ -1,33 +1,235 @@
-# Get object aws_vpc by vpc_id
-data "aws_vpc" "default" {
-  count = local.enabled ? 1 : 0
-  id    = var.vpc_id
+# Process all the optional inputs into a fully fleshed out configuration
+# the rest of the code can consume.
+
+locals {
+  enabled = module.this.enabled && (var.public_subnets_enabled || var.private_subnets_enabled) && (var.ipv4_enabled || var.ipv6_enabled)
+
+  # We are going to reference `enabled` a *lot*, so abbreviate it
+  e = local.enabled
+
+  # Use delimiter shortcut for creating AZ-based Names/IDs
+  delimiter = module.this.delimiter
+
+  # The RFC 6052 "well known" NAT64 CIDR
+  nat64_cidr = "64:ff9b::/96"
+
+  # In case we later decide to compute it
+  vpc_id = var.vpc_id
+
+  #####################################################################
+  ## Determine the set of availability zones in which to deploy subnets
+  #  Priority is
+  #  - availability_zone_ids
+  #  - availability_zones
+  #  - data.aws_availability_zones.default
+
+  use_az_ids      = local.e && length(var.availability_zone_ids) > 0
+  use_az_var      = local.e && length(var.availability_zones) > 0
+  use_default_azs = local.e && !(local.use_az_ids || local.use_az_var)
```

Comment:
I am not concerned about having unused local variables. They serve as documentation and maintain parallel equivalence (ipv4 vs ipv6) making the code easier to understand and modify in the future. 

In this particular case `use_default_azs` is not used because it is the only remaining choice in the conditional testing for the other 2 conditions.